core/primitives/src/shard_layout.rs

```diff
@@ -245,13 +262,16 @@ impl TryFrom<&[u8]> for ShardUId {
         Ok(Self { version, shard_id })
     }
 }
```

Comment:
Unrelated to the PR, but, rather than `TryFrom<&[u8]>` I'd probably expect to see 

```rust
impl ShardUid {
  fn to_bytes(&self) -> [u8; 8] {}
  fn from_bytes(bytes: [u8; 8]) -> ShardId {}
}
```

No need to return a runtime error if the precondition can be expressed as a type.